File: Comparator/IntegerComparator.java
package Comparator;


import usertype.IntegerClass;

import java.io.Serializable;

public class IntegerComparator implements Comparator, Serializable {
    @Override
    public int compare(Object o1, Object o2) {
        return ((IntegerClass)o1).getValue() - ((IntegerClass)o2).getValue();
    }
}

File: Comparator/IntegerComparator.kt
package Comparator

import usertype.IntegerClass
import java.io.Serializable

class IntegerComparator : Comparator, Serializable {
    override fun compare(o1: Any?, o2: Any?): Int {
        return (o1 as IntegerClass?)!!.value - (o2 as IntegerClass?)!!.value
    }
}